Sources/SourceKitLSP/Swift/MacroExpansion.swift

```diff
@@ -74,51 +74,17 @@ extension SwiftLanguageService {
     var macroExpansionFilePaths: [URL] = []
     for macroEdit in expansion.edits {
       if let bufferName = macroEdit.bufferName {
-        // buffer name without ".swift"
-        let macroExpansionBufferDirectoryName =
-          bufferName.hasSuffix(".swift")
-          ? String(bufferName.dropLast(6))
-          : bufferName
+        let macroExpansionReferenceDocument = MacroExpansionReferenceDocument(
+          macroExpansionEditRange: macroEdit.range,
+          sourceFileURL: sourceFileURL,
+          selectionRange: expandMacroCommand.positionRange,
+          bufferName: bufferName
+        )
 
-        let macroExpansionBufferDirectoryURL = self.generatedMacroExpansionsPath
-          .appendingPathComponent(macroExpansionBufferDirectoryName)
+        macroExpansionFilePaths.append(try macroExpansionReferenceDocument.url)
```

Comment:
These are no longer file paths, right? So, this should probably be `macroReferenceDocuments`. Also, we could maintain a little bit more type information here and make this a variable of type `[ReferenceDocumentURL]`.